test/modules/TestResultCoordinator/Reports/DeploymentTestReportGenerator.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    sealed class DeploymentTestReportGenerator : ITestResultReportGenerator
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(DeploymentTestReportGenerator));
+
+        readonly string trackingId;
+
+        public DeploymentTestReportGenerator(
+            string trackingId,
+            string expectedSource,
+            ITestResultCollection<TestOperationResult> expectedTestResults,
+            string actualSource,
+            ITestResultCollection<TestOperationResult> actualTestResults)
+        {
+            this.trackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
+            this.ExpectedTestResults = Preconditions.CheckNotNull(expectedTestResults, nameof(expectedTestResults));
+            this.ExpectedSource = Preconditions.CheckNonWhiteSpace(expectedSource, nameof(expectedSource));
+            this.ActualSource = Preconditions.CheckNonWhiteSpace(actualSource, nameof(actualSource));
+            this.ActualTestResults = Preconditions.CheckNotNull(actualTestResults, nameof(actualTestResults));
+
+            this.TestResultComparer = new DeploymentTestResultComparer();
+            this.ResultType = TestOperationResultType.Deployment.ToString();
+        }
+
+        internal string ActualSource { get; }
+
+        internal ITestResultCollection<TestOperationResult> ActualTestResults { get; }
+
+        internal string ExpectedSource { get; }
+
+        internal ITestResultCollection<TestOperationResult> ExpectedTestResults { get; }
+
+        internal string ResultType { get; }
+
+        internal ITestResultComparer<TestOperationResult> TestResultComparer { get; }
+
+        /// <summary>
+        /// Compare 2 data stores and counting expected, actual, and matched results; and return a deployment test report.
+        /// Actual deployment results can be less than expected deployment results.
+        /// An actaul deployment test result is possible to use for verification of more than 1 expected deplpyment test result.
+        /// It will log fail if actual store has more results than expect store.
+        /// </summary>
+        /// <returns>Test Result Report.</returns>
+        public async Task<ITestResultReport> CreateReportAsync()
```

Comment:
I would like to talk to you to understand how it can be determined if the test passed or failed from the report.

Reply:
Explained in person